platforms/ide/tooling-api-builders/src/main/java/org/gradle/tooling/internal/provider/runner/ProblemsProgressEventConsumer.java

```diff
@@ -79,29 +86,48 @@ public void progress(OperationIdentifier buildOperationId, OperationProgressEven
         Object details = progressEvent.getDetails();
         if (details instanceof DefaultProblemProgressDetails) {
             Problem problem = ((DefaultProblemProgressDetails) details).getProblem();
-            eventConsumer.progress(
-                new DefaultProblemEvent(
-                    new DefaultProblemDescriptor(
-                        new OperationIdentifier(
-                            idFactory.nextId()
-                        ),
-                        buildOperationId),
-                    new DefaultProblemDetails(
-                        toInternalCategory(problem.getProblemCategory()),
-                        toInternalLabel(problem.getLabel()),
-                        toInternalDetails(problem.getDetails()),
-                        toInternalSeverity(problem.getSeverity()),
-                        toInternalLocations(problem.getLocations()),
-                        toInternalDocumentationLink(problem.getDocumentationLink()),
-                        toInternalSolutions(problem.getSolutions()),
-                        toInternalAdditionalData(problem.getAdditionalData()),
-                        problem.getException()
-                    )
-                )
-            );
+            eventConsumer.progress(createProblemEvent(buildOperationId, problem));
```

Comment:
I think we should model the summary as a collection of problems here. Then TAPI protocol could send either a single problem or a collection of problems in one event. Note, that `InternalBasicProblemDetails` has not been released, therefore we are free to change that to our liking. We might want to introduce something like `InternalProblemCollectionDetails` (name can be be improved)